File: problems/1614_maximum_nesting_depth_of_the_parentheses.py
class Solution:
    def maxDepth(self, s: str) -> int:
        depth = 0
        max_depth = 0
        for c in s:
            match c:
                case "(":
                    depth += 1
                    max_depth = max(max_depth, depth)
                case ")":
                    depth -= 1
        return max_depth

File: problems/1614-maximum-nesting-depth-of-the-parentheses.ts
function maxDepth(s: string): number {
  let depth = 0
  let maxDepth = 0
  for (const char of s) {
    switch (char) {
      case "(":
        depth++
        maxDepth = Math.max(maxDepth, depth)
        break
      case ")":
        depth--
        break
    }
  }
  return maxDepth
}

maxDepth("(1+(2*3)+((8)/4))+1")
